storage/pebble/register.go

```diff
@@ -0,0 +1,154 @@
+package pebble
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/onflow/atree"
+
+	errs "github.com/onflow/flow-evm-gateway/models/errors"
+)
+
+var _ atree.Ledger = &Register{}
+
+type Register struct {
+	height uint64
+	store  *Storage
+	mux    sync.RWMutex
+}
+
+func NewRegister(store *Storage, height uint64) *Register {
+	return &Register{
+		store:  store,
+		height: height,
+		mux:    sync.RWMutex{},
+	}
+}
+
+func (l *Register) GetValue(owner, key []byte) ([]byte, error) {
+	l.mux.RLock()
+	defer l.mux.RUnlock()
+
+	iter, err := l.store.db.NewIter(&pebble.IterOptions{
+		LowerBound: l.idLower(owner, key),
+		UpperBound: l.idUpper(owner, key),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create register range itterator: %w", err)
+	}
+	defer iter.Close()
```

Comment:
```go
defer func() {
		err := iterator.Close()
		if err != nil {
			r.store.log.Error().Err(err).Msg("failed to close register iterator")
		}
	}()
```
It would be nice to handle the possible error and add some logging, if possible.